### PR TITLE
Channel: Add unit tests for AbstractChannel edge-case failures and buffer releases

### PR DESCRIPTION
**Pull request by rajan-github**
**Motivation:**
Core channel lifecycle paths in AbstractChannel (such as writing to a channel closed by an I/O error, flushing an unconnected open channel, and flushing a closed channel) lacked explicit unit test coverage for proper exception propagation and memory leak prevention.

**Modifications:**
Added unit tests to AbstractChannelTest covering:
- Writing to a channel closed via a write-error I/O exception to ensure proper ClosedChannelException propagation and buffer release.
- Flushing an unconnected but open channel to verify NotYetConnectedException handling and resource cleanup.
- Flushing an unconnected channel that subsequently closes to verify proper ClosedChannelException mapping and reference count management.

**Result:**
Improved test coverage and robustness for edge-case failure modes and resource-leak protections in AbstractChannel.

### CODE REVIEW

File: transport/src/test/java/io/netty/channel/AbstractChannelTest.java [hunk 1]
```diff
@@ -166,6 +170,131 @@ public boolean isActive() {
...
+        EventLoop loop = new DefaultEventLoop();
+        try {
+            registerChannel(loop, channel);
+            channel.connect(new InetSocketAddress(NetUtil.LOCALHOST, 8888)).sync();
+
+            assertSame(ioException, channel.writeAndFlush("").await().cause());
+            ByteBuf msg = Unpooled.wrappedBuffer(new byte[16]);
+            assertEquals(1, msg.refCnt());
+            ChannelPromise promise = channel.newPromise();
+            channel.unsafe().write(msg, promise);
```
Comment (normanmaurer) [inline transport/src/test/java/io/netty/channel/AbstractChannelTest.java]: This needs to be executed in the event loop thread as otherwise its not thread-safe and violates the model

File: transport/src/test/java/io/netty/channel/AbstractChannelTest.java [hunk 2]
```diff
@@ -166,6 +170,131 @@ public boolean isActive() {
...
+
+            assertTrue(promise.isDone());
+            assertInstanceOf(ClosedChannelException.class, promise.cause());
+            assertEquals(0, msg.refCnt());
+        } finally {
+            channel.close();
```
Comment (normanmaurer) [inline transport/src/test/java/io/netty/channel/AbstractChannelTest.java]:
```suggestion
            channel.close().sync();
```

File: transport/src/test/java/io/netty/channel/AbstractChannelTest.java (hunk 1 above)
Comment (normanmaurer) [reply transport/src/test/java/io/netty/channel/AbstractChannelTest.java]: This is true for any operation that is done via `unsafe()` in the changes done here.

**Review by normanmaurer — CHANGES_REQUESTED** [review]

File: transport/src/test/java/io/netty/channel/AbstractChannelTest.java (hunk 1 above)
Comment (rajan-github) [reply transport/src/test/java/io/netty/channel/AbstractChannelTest.java]: @normanmaurer Thanks for catching that. I've updated all the new test cases to explicitly wrap the `unsafe()` operations (write, flush, and close) inside `eventLoop.execute(...)` blocks to strictly adhere to Netty's thread-safety model.

Please let me know if this looks good or if any further adjustments are needed. Thanks again for the guidance!

File: transport/src/test/java/io/netty/channel/AbstractChannelTest.java (hunk 2 above)
Comment (rajan-github) [reply transport/src/test/java/io/netty/channel/AbstractChannelTest.java]: Thanks for catching that. It has been applied in the changes now.
